cardinal/chain/adapter.go

```diff
@@ -17,8 +18,18 @@ import (
 
 // Adapter is a type that helps facilitate communication with the EVM base shard.
 type Adapter interface {
-	Submit(ctx context.Context, namespace string, tick uint64, txs []byte) error
-	QueryBatches(ctx context.Context, req *shardtypes.QueryBatchesRequest) (*shardtypes.QueryBatchesResponse, error)
+	Writer
+	Reader
+}
+
+type Writer interface {
+	Submit(ctx context.Context, p *sign.SignedPayload, txID, tick uint64) error
+}
+
+type Reader interface {
```

Comment:
nit: can we call this ShardAdapterReader

Reply:
if we go with renaming the package to `shard` i think we can forego the shard prefix there and possibly just do `ReadAdapter`

Reply:
sounds good